Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -929,7 +929,54 @@ private void DestroyAllServices()
         {
             if (!ExecuteOnAllServices(service => service.Destroy())) { return; }
 
+            List<Type> serviceTypes = activeSystems.Keys.ToList<Type>();
+            foreach (Type type in serviceTypes)
+            {
+                if (typeof(IMixedRealityBoundarySystem).IsAssignableFrom(type))
+                {
+                    UnregisterService<IMixedRealityBoundarySystem>();
+                    boundarySystem = null;
```

Comment:
I'm not sure if this PR fully fixes #4579. It seems like this null setting take place within the `UnregisterService` call itself, instead of just when we're destroying every service.

Reply:
good catch. i misread 4579 when fixing :)